packages/logging/service_test.go

```diff
@@ -324,3 +340,199 @@ func Test_QueryForDogu(t *testing.T) {
 		assert.ErrorContains(t, err, assert.AnError.Error())
 	})
 }
+
+func TestLoggingService_SetLogLevel(t *testing.T) {
+	tests := []struct {
+		name          string
+		req           *pb.LogLevelRequest
+		xResponse     bool
+		xResponseCode codes.Code
+	}{
+		{
+			name: "Set LogLevel DEBUG",
+			req: &pb.LogLevelRequest{
+				DoguName: "test",
+				LogLevel: pb.LogLevel_DEBUG,
+			},
+			xResponse:     true,
+			xResponseCode: codes.OK,
+		},
+		{
+			name: "Set LogLevel INFO",
+			req: &pb.LogLevelRequest{
+				DoguName: "test",
+				LogLevel: pb.LogLevel_INFO,
+			},
+			xResponse:     true,
+			xResponseCode: codes.OK,
+		},
+		{
+			name: "Set LogLevel WARN",
+			req: &pb.LogLevelRequest{
+				DoguName: "test",
+				LogLevel: pb.LogLevel_WARN,
+			},
+			xResponse:     true,
+			xResponseCode: codes.OK,
+		},
+		{
+			name: "Set LogLevel ERROR",
+			req: &pb.LogLevelRequest{
+				DoguName: "test",
+				LogLevel: pb.LogLevel_ERROR,
+			},
+			xResponse:     true,
+			xResponseCode: codes.OK,
+		},
+		{
+			name: "Set wrong LogLevel",
+			req: &pb.LogLevelRequest{
+				DoguName: "test",
+				LogLevel: 4,
+			},
+			xResponse:     false,
+			xResponseCode: codes.InvalidArgument,
+		},
+		{
+			name: "Empty dogu name",
+			req: &pb.LogLevelRequest{
+				DoguName: "",
+				LogLevel: pb.LogLevel_DEBUG,
+			},
+			xResponse:     false,
+			xResponseCode: codes.InvalidArgument,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			mockedLogProvider := newMockLogProvider(t)
+			mockedConfigProvider := newMockConfigProvider(t)
+			mockedDoguRestarter := newMockDoguRestarter(t)
+			mockedConfigurationContext := NewMockConfigurationContext(t)
+
+			if tc.xResponse {
+				mockedConfigProvider.On("DoguConfig", mock.Anything).Return(mockedConfigurationContext)
+
+				mockedConfigurationContext.On("Get", mock.Anything).Return("", nil)
+				mockedConfigurationContext.On("Set", mock.Anything, mock.Anything).Return(nil)
+
+				mockedDoguRestarter.On("RestartDogu", mock.Anything, mock.Anything).Return(nil)
+			}
+
+			sut := NewLoggingService(mockedLogProvider, mockedConfigProvider, mockedDoguRestarter)
+
+			resp, err := sut.SetLogLevel(context.TODO(), tc.req)
+
+			assert.Equal(t, tc.xResponse, resp != nil)
+			assert.Equal(t, tc.xResponseCode, status.Code(err))
+
+			if tc.xResponse {
+				mockedConfigProvider.AssertNumberOfCalls(t, "DoguConfig", 1)
+				mockedConfigurationContext.AssertNumberOfCalls(t, "Get", 1)
+				mockedConfigurationContext.AssertNumberOfCalls(t, "Set", 1)
+				mockedDoguRestarter.AssertNumberOfCalls(t, "RestartDogu", 1)
+			} else {
+				mockedConfigProvider.AssertNotCalled(t, "DoguConfig", mock.Anything)
+				mockedConfigurationContext.AssertNotCalled(t, "Get", mock.Anything)
+				mockedConfigurationContext.AssertNotCalled(t, "Set", mock.Anything, mock.Anything)
+				mockedDoguRestarter.AssertNotCalled(t, "RestartDogu", mock.Anything, mock.Anything)
+			}
```

Comment:
I understand what you are doing here. And yet, is this code not the same as `mock.AssertExpectations()`?

If so, you can remove the whole code block because `t.Cleanup()` will run the expectation asserting during the test cleanup. From what I learned even inside of capsulated tests with `t.Run()`.

This counts for similar cases in the whole test file.